cli/src/semgrep/config_resolver.py

```diff
@@ -805,6 +898,19 @@ def url_for_policy() -> str:
     return f"{env.semgrep_url}/{DEFAULT_SEMGREP_APP_CONFIG_URL}?{params_str}"
 
 
+PRODUCT_NAMES = {
```

Comment:
The developer ergonomics for interacting with the set of products we supports seems rusty.

#### `rule_match.py`
```python
# TODO: diff with rule.py product() method?
@property
def product(self) -> out.Product:
    if self.metadata.get("product") == "secrets":
        return out.Product(out.Secrets())
    elif "sca_info" in self.extra:
        return out.Product(out.SCA())
    else:
        return out.Product(out.SAST())
```

#### `rule.py`
```python
@property
def product(self) -> out.Product:
    if "r2c-internal-project-depends-on" in self._raw:
        return out.Product(out.SCA())
    elif "product" in self.metadata:
        if self.metadata.get("product") == "secrets":
            return out.Product(out.Secrets())
        else:
            return out.Product(out.SAST())
    else:
        return out.Product(out.SAST())
```

Reply:
Yeah, 100% theres opportunity to improve this.
Some useful context on this PR is that I threw this together in 30min for @salolivares to pick up and run with. I shared a bunch of notes with him about things that need improving and we can totally add this to the list (although it does feel a bit like scope creep)

Reply:
Going to defer this: don't have a good enough understanding of the CLI yet + don't wanna balloon the scope of this PR. Is there a good linear project I can throw this improvement in?

Reply:
feel free to resolve this conversation!